.buildkite/pipeline.yml

```diff
@@ -2,338 +2,335 @@ steps:
     concurrency_group: 'browserstack-app'
 
-  - block: "Trigger full test suite"
-
-  - label: ':android: Android 5 end-to-end tests'
-    skip: Temporarily disabled due to flakiness with BrowserStack tunneling
+  - label: ':android: Android 7 end-to-end tests'
     timeout_in_minutes: 60
     plugins:
       artifacts#v1.2.0:
         download: "build/fixture.apk"
-      docker-compose#v3.1.0:
+      docker-compose#v3.3.0:
         run: android-maze-runner
     env:
-      DEVICE_TYPE: "ANDROID_5_0"
+      DEVICE_TYPE: "ANDROID_7_1"
       APP_LOCATION: "/app/build/fixture.apk"
     concurrency: 10
     concurrency_group: 'browserstack-app'
 
-  - label: ':android: Android 6 end-to-end tests'
+  - label: ':android: Android 8.0 end-to-end tests'
     timeout_in_minutes: 60
     plugins:
       artifacts#v1.2.0:
         download: "build/fixture.apk"
-      docker-compose#v3.1.0:
+      docker-compose#v3.3.0:
         run: android-maze-runner
     env:
-      DEVICE_TYPE: "ANDROID_6_0"
+      DEVICE_TYPE: "ANDROID_8_0"
       APP_LOCATION: "/app/build/fixture.apk"
     concurrency: 10
     concurrency_group: 'browserstack-app'
 
-  - label: ':android: Android 7 end-to-end tests'
+  - label: ':android: Android 8.1 end-to-end tests'
     timeout_in_minutes: 60
     plugins:
       artifacts#v1.2.0:
         download: "build/fixture.apk"
-      docker-compose#v3.1.0:
+      docker-compose#v3.3.0:
         run: android-maze-runner
     env:
-      DEVICE_TYPE: "ANDROID_7_1"
+      DEVICE_TYPE: "ANDROID_8_1"
       APP_LOCATION: "/app/build/fixture.apk"
     concurrency: 10
     concurrency_group: 'browserstack-app'
+    soft_fail:
+      - exit_status: "*"
 
-  - label: ':android: Android 8.0 end-to-end tests'
+  - label: ':android: Android 9 end-to-end tests'
     timeout_in_minutes: 60
     plugins:
-        artifacts#v1.2.0:
-            download: "build/fixture.apk"
-        docker-compose#v2.6.0:
-            run: android-maze-runner
+      artifacts#v1.2.0:
+        download: "build/fixture.apk"
+      docker-compose#v3.3.0:
+        run: android-maze-runner
     env:
-        DEVICE_TYPE: "ANDROID_8_0"
-        APP_LOCATION: "/app/build/fixture.apk"
+      DEVICE_TYPE: "ANDROID_9_0"
+      APP_LOCATION: "/app/build/fixture.apk"
     concurrency: 10
     concurrency_group: 'browserstack-app'
 
-  - label: ':android: Android 8.1 end-to-end tests'
+  - label: ':android: Android 10 end-to-end tests'
     timeout_in_minutes: 60
     plugins:
       artifacts#v1.2.0:
         download: "build/fixture.apk"
-      docker-compose#v3.1.0:
+      docker-compose#v3.3.0:
         run: android-maze-runner
     env:
-      DEVICE_TYPE: "ANDROID_8_1"
+      DEVICE_TYPE: "ANDROID_10_0"
       APP_LOCATION: "/app/build/fixture.apk"
     concurrency: 10
     concurrency_group: 'browserstack-app'
     soft_fail:
-        - exit_status: "*"
+      - exit_status: "*"
 
-  - label: ':android: NDK 16b SDK 4.4 Instrumentation tests'
+  - label: ':android: Linter'
```

Comment:
Yes, that's basically the case, although depending on which agent gets assigned to the step there may be. a bit of time needed to pull the image to it.  Looking at this I also noticed that the image-repository was not set on one of the build steps, but that actually we didn't need the pre-build at all (because it's only run once and run will build it if needed).

I've ended end removing a build step, moving a few around, and have reviewed the timeouts in general (tweaking the instrumentation test steps down from 60 mins to 30).